client/post-editor/editor-media-advanced/index.jsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External depencencies
+ */
+import React, { PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { toggleEditorMediaAdvanced } from 'state/ui/editor/media/actions';
+import Dialog from 'components/dialog';
+
+function EditorMediaAdvanced( { visible, toggleVisible } ) {
+	return (
+		<Dialog isVisible={ visible } onClose={ toggleVisible }>
+			WIP
+		</Dialog>
+	);
+}
+
+EditorMediaAdvanced.propTypes = {
+	visible: PropTypes.bool,
+	toggleVisible: PropTypes.func
+};
+
+EditorMediaAdvanced.defaultProps = {
+	visible: false,
+	toggleVisible: () => {}
+};
```

Comment:
>why put these here when we are already using destructuring in the function declaration?

You know, that's an interesting question. One has to wonder why `defaultProps` was [one of the two properties](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#stateless-functional-components) that the React team chose to support for stateless functional components. Perhaps there's some semantics of `defaultProps` that wouldn't be reflected with the destructured default values? I couldn't find much discussion on the topic, though [this commenter](https://medium.com/@bernhard/great-article-c2a5c9c8007e#.rezgybah5) appears to prefer default arguments from a function purity perspective. On the flip-side, it does make the function declaration line a bit longer.

Do you have a strong preference either way? I'm inclined to keep it as is, mostly for aesthetics to this last point and for proximity to the `propTypes` declaration for quick scanning. Additionally, it may be that this is easier for static analysis if we choose to pursue some automated documentation of components.